gprofiler/profilers/java.py

```diff
@@ -114,6 +115,23 @@ def get_ap_log(self) -> str:
         return self._ap_log
 
 
+class AsyncProfiledProcessMonitor:
+    def __init__(self):
+        self._attached_processes: List[int] = []
```

Comment:
You're now tracking processes both here and in `JavaProfiler._profiled_processes` :(